src/main/java/graphql/annotations/GraphQLAnnotations.java

```diff
@@ -495,24 +496,33 @@ private boolean checkIfPrefixGetterExists(Class c, String prefix, String propert
         return true;
     }
 
-    private GraphQLOutputType getGraphQLConnection(boolean isConnection, AccessibleObject field, GraphQLOutputType type, GraphQLOutputType outputType, GraphQLFieldDefinition.Builder builder) {
-        if (isConnection) {
-            if (type instanceof GraphQLList) {
-                graphql.schema.GraphQLType wrappedType = ((GraphQLList) type).getWrappedType();
-                assert wrappedType instanceof GraphQLObjectType;
-                String annValue = field.getAnnotation(GraphQLConnection.class).name();
-                String connectionName = annValue.isEmpty() ? wrappedType.getName() : annValue;
-                GraphQLObjectType edgeType = RELAY_TYPES.edgeType(connectionName, (GraphQLOutputType) wrappedType, null, Collections.<GraphQLFieldDefinition>emptyList());
-                outputType = RELAY_TYPES.connectionType(connectionName, edgeType, Collections.emptyList());
-                builder.argument(RELAY_TYPES.getConnectionFieldArguments());
-            }
+    private GraphQLOutputType getGraphQLConnection(AccessibleObject field, GraphQLOutputType type, GraphQLFieldDefinition.Builder builder) {
+        boolean isNonNull = (type instanceof GraphQLNonNull);
+        if (isNonNull) {
+            type = (GraphQLOutputType) ((GraphQLNonNull) type).getWrappedType();
+        }
+
+        graphql.schema.GraphQLType wrappedType = ((GraphQLList) type).getWrappedType();
+        assert wrappedType instanceof GraphQLObjectType;
+        String annValue = field.getAnnotation(GraphQLConnection.class).name();
```

Comment:
I would call it connectionName, and instead of initializing a new variable next line, would use it.

Reply:
done